### ISSUE TITLE
donating in IOS +Firefox night mode = can't see what you're typing

### ISSUE THREAD

**Issue by WillatMozFdn** [mozilla/donate-wagtail#1139]
not sure if this is a known bug elsewhere, but a donor alerted us to this, and Stephanie L has confirmed ... if you're using Firefox night mode on an iOS device, the credit card numbers (and exp date/CVV code) do not display at all as you type them.  Screenshot from Stephanie's phone:  

![image](https://user-images.githubusercontent.com/23637882/92517518-cb11cb80-f1e4-11ea-9db4-fb028dd736a6.png)

(this was with all credit card info typed in.)

**Comment by cadecairos** [comment]
Nice find, we'll take care of this in our 9/15 sprint

**Comment by mmmavis** [comment]
These fields are injected by Braintree. Each of them is actually an iframe so we have no control over it at all. Unfortunately, we won't be able to fix this issue on our end.

**Comment by TheoChevalier** [comment]
Looks like we already have a hacky way to style these fields (I remembered we had an issue early in the project with the text being too small in these fields), I’m wondering if we could leverage this?

https://github.com/mozilla/donate-wagtail/blob/master/source/sass/components/_form-item.scss#L107

**Comment by mmmavis** [comment]
@TheoChevalier thanks for the code reference! However, the selector `.form-item__dummy-placeholder` is just for the wrapper `div` outside of the iframe. So even if we add SCSS rules there, they won't apply to elements in the iframe.

**Comment by cadecairos** [comment]
It looks like we can [style the hosted fields using JavaScript](https://developers.braintreepayments.com/guides/hosted-fields/styling/javascript/v3)

[Here's where we call this method in our code](https://github.com/mozilla/donate-wagtail/blob/caa1a065d72c55622402146e8837ac4afb814653/source/js/payments-card.js#L63-L85)

**Comment by jamilasnell** [comment]
Let's verify if this is still an issue

**Comment by danielfmiranda** [comment]
Can confirm that this is still an issue!

**Comment by danielfmiranda** [comment]
Hi Everyone! I have found two possible solutions and was wondering if I can have some feedback from the design team @kristinashu to see which one we would prefer to move forward with. 

The first solution, which is probably the closest we can get to the desired behavior, is applying a filter to invert the colors of the iframe form that Mavis mentioned above. This is the result on IOS firefox with dark mode:

<img width="400" alt="" src="https://user-images.githubusercontent.com/18314510/148297069-d91c4b22-61cd-4dd5-822d-c3b3aae2135a.PNG">

However, this then introduces a interesting possible side-effect for users using extensions such as one password. Since onepassword renders a button within the iframe, this fix then renders the button as a solid color:

<img width="928" alt="Screen Shot 2022-01-05 at 1 58 16 PM" src="https://user-images.githubusercontent.com/18314510/148297738-f54e9a88-1b5e-48dc-a2e8-712f27290938.png">

I have tried a number of different ways to reverse this, but as Mavis mentioned above, the fact that this is being rendered in an iframe from the braintree side, I have been having a difficult time.




This then brings me to the second solution, which is the one I am leaning towards the most, which is updating the background color for these fields as white, so the user can see what they are typing. This is my preferred solution as it solves the issue for firefox IOS users on dark mode, but then does not make any changes for users accessing the site using any other medium, which I assume is a majority of visitors.

If the design team would like, I can also update the fields above the credit card info to a white background as well, so the all items on the form appear uniform instead of just the credit card fields rendering with a white background for firefox IOS dark mode users. 

<img width="400" alt="" src="https://user-images.githubusercontent.com/18314510/148297722-9154b886-cce0-4b71-b2c8-606070760581.PNG">


Thank you!
